### PR TITLE
Document 'dart pub add' support for adding dependency overrides

### PR DESCRIPTION
**Pull request by shaikahmadnawaz**
Hey there! I've created a PR that adds documentation for the new functionality of dart pub add to add dependency overrides. The documentation is now available on the v3.0 branch. Please take a look and let me know if you have any feedback or suggestions. Thank you!

### CODE REVIEW

**Comment by domesticmouse** [comment]
Please sign the CLA. We can't review this PR until you've signed the CLA.

File: src/tools/pub/cmd/pub-add.md
Comment (domesticmouse) [inline src/tools/pub/cmd/pub-add.md:64]: I believe these spaces are intended, please revert.

File: src/tools/pub/cmd/pub-add.md
Comment (domesticmouse) [inline src/tools/pub/cmd/pub-add.md:170]: Likewise

File: src/tools/pub/cmd/pub-add.md
Comment (domesticmouse) [inline src/tools/pub/cmd/pub-add.md:195]: And again

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: src/tools/pub/cmd/pub-add.md
Comment (shaikahmadnawaz) [reply src/tools/pub/cmd/pub-add.md:195]: > And again

Other than these indentations is everything okay?

File: src/tools/pub/cmd/pub-add.md
Comment (domesticmouse) [reply src/tools/pub/cmd/pub-add.md:170]: That's up to @parlough =)

**Review by shaikahmadnawaz — COMMENTED** [review]
Indentations reverted, sorry it happened because of my VS Code extension

**Comment by parlough** [comment]
Hey @shaikahmadnawaz, thanks so much for working on this! Looks to be a great start.

I don't have a chance to review this today, but I plan to get to it tomorrow. Sorry about the delay!

**Comment by shaikahmadnawaz** [comment]
> Hey @shaikahmadnawaz, thanks so much for working on this! Looks to be a great start.
> 
> I don't have a chance to review this today, but I plan to get to it tomorrow. Sorry about the delay!

Thank you for letting me know, I'm waiting for your feedback 😊

**Review by domesticmouse — APPROVED** [review]
LGTM, pending @parlough's review

File: src/tools/pub/cmd/pub-add.md
Comment (parlough) [inline src/tools/pub/cmd/pub-add.md:45]: This is pretty good. I think we can drop the first sentence though, since I don't think users are familiar with the new way to add dev dependency yet anyway. 

Then we want to link to the dependency override documentation, so they can learn the specific details and potential concerns with them. That way we don't have to repeat too much here. The dev dependency section does the same thing.

Also we need to be careful as users don't necessarily need to specify a version, they could specify any of the [source descriptors](https://dart.dev/tools/pub/cmd/pub-add#source-descriptor). How about something like this:

```suggestion
To specify a [dependency override][], add the `override:` prefix and
specify any [source descriptor](#source-descriptor).

[dependency override]: /tools/pub/dependencies#dependency-overrides
```

File: src/tools/pub/cmd/pub-add.md
Comment (parlough) [inline src/tools/pub/cmd/pub-add.md:47]: Good example. A few things here:

- We break up lines >= 80 chars long up with something called [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks)
- Consider making "For example" bold, since the example might be enough for some readers.
- I think "add an override for x package version" might be a bit confusing, as a dependency override doesn't override the specified package version, it specifies that version as the override.

How does this sound?

```suggestion
**For example**, to override all references to `package:foo`
to use version `1.0.0` of the package,
run the following command:
```

File: src/tools/pub/cmd/pub-add.md
Comment (parlough) [inline src/tools/pub/cmd/pub-add.md:49]: When adding code blocks, we try to add a language or type, in this case `terminal` to allow either syntax highlighting or special styles. For example, on dart.dev, terminal code blocks are given a black background.

````suggestion
```terminal
````

File: src/tools/pub/cmd/pub-add.md
Comment (parlough) [inline src/tools/pub/cmd/pub-add.md:50]: On dart.dev, we add a $ before a command to make it more clear that it is a command:

```suggestion
$ dart pub add override:foo:1.0.0
```

File: src/tools/pub/cmd/pub-add.md
Comment (parlough) [inline src/tools/pub/cmd/pub-add.md:53]: On dart.dev, we try to use [present tense](https://developers.google.com/style/tense).

```suggestion
This adds the override to your `pubspec.yaml` file:
```

File: src/tools/pub/cmd/pub-add.md
Comment (parlough) [inline src/tools/pub/cmd/pub-add.md:55]: Adding the `yaml` specifier to the code block enables syntax highlighting.

````suggestion
```yaml
````

File: src/tools/pub/cmd/pub-add.md
Comment (parlough) [inline src/tools/pub/cmd/pub-add.md:57]: In YAML the indentation is important here:

```suggestion
  foo: 1.0.0
```

**Review by parlough — CHANGES_REQUESTED** [review]
Thanks for the pull request @shaikahmadnawaz and for your patience! 

It looks pretty good overall, just needs some modification and cleanup to align with our standards and best practices. Also some cross-links to elsewhere to provide some extra context and reduce duplication.

I left suggestions and context for the suggestions. If you're happy with the suggestions, you can commit them directly from the GitHub UI. Feel free to make other changes or comment on my replies for more discussion. And always feel free to ask questions. 

Thanks again!

**Comment by shaikahmadnawaz** [comment]
> Thanks for the pull request @shaikahmadnawaz and for your patience!
> 
> It looks pretty good overall, just needs some modification and cleanup to align with our standards and best practices. Also some cross-links to elsewhere to provide some extra context and reduce duplication.
> 
> I left suggestions and context for the suggestions. If you're happy with the suggestions, you can commit them directly from the GitHub UI. Feel free to make other changes or comment on my replies for more discussion. And always feel free to ask questions.
> 
> Thanks again!

Thank you for your feedback and suggestions! I have reviewed your comments and made the necessary updates to align with your standards and best practices. Please let me know if there's anything else I can do to improve this PR.

**Review by parlough — APPROVED** [review]
Thanks for the pull request @shaikahmadnawaz and making those adjustments, looks good to me!

**Comment by shaikahmadnawaz** [comment]
> Thanks for the pull request @shaikahmadnawaz and making those adjustments, looks good to me!

Thank you so much @parlough for your positive feedback and for taking the time to review my pull request. I'm really glad to hear that the changes I made look good to you! This is my first contribution to this repository, and I'm thrilled to have been able to make a meaningful contribution. Your encouragement means a lot to me, and I look forward to contributing more in the future.

**Pull request merged**
